instrumentation/spring/spring-boot-actuator-autoconfigure-2.0/javaagent/build.gradle.kts

```diff
@@ -22,6 +22,16 @@ dependencies {
   latestDepTestLibrary("ch.qos.logback:logback-classic:+")
 }
 
+testing {
+  suites {
+    val testPrometheus by registering(JvmTestSuite::class) {
+      dependencies {
+        runtimeOnly("io.micrometer:micrometer-registry-prometheus:1.14.3")
+      }
+    }
+  }
+}
```

Comment:
what's the failure for this test without your change to `OpenTelemetryMeterRegistry`?

Reply:
just tried... there is no failure - which confirms that the error does not always appear

but I've improved the test

Reply:
what's the failure now (without your change to `OpenTelemetryMeterRegistry`)?

Reply:
@trask the test is still passing with the unchanged code

I don't know how to force spring to use a different resolution order from the provided constraints

Reply:
the test is still passing the old code - but I don't have an idea how to make a better test - you'd have to force spring to randomize the bean order within the provided constraints

Reply:
thanks, makes sense